examples/connext_dds/xml_application_env/c++11/application.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0"?>
+
+<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+     xsi:noNamespaceSchemaLocation="http://community.rti.com/schema/6.0.0/rti_dds_profiles.xsd"
+     version="6.0.0">
+  
+  <!-- QoS Library containing the QoS profile used in the example. 
+          KeepLastReliable is used as the base profile. -->
+  <qos_library name="Qos_Library">
+    <qos_profile name="DefaultProfile">
+	      <participant_qos base_name="BuiltinQosLibExp::Generic.KeepLastReliable">
+        <!-- The participant name, if it is set, will be displayed in the RTI
+                    Analyzer tool, making it easier for you to tell one application
+                    from another when you're debugging. -->
+        <participant_name>
+          <name>XML Application Creation C++11 Example</name>
+        </participant_name>
+      </participant_qos>
+    </qos_profile>
+
+  </qos_library>
+  
+  <!-- Define the available types which can be realized as DynamicData.
+          The type used in this example is the KeyValuePair.  It represents a 
+          key and a value to represent an envirnment variable.  Additionally 
+          an id is used to identify different instances of the application,
+          as well as a time value to mark when the variables were published. -->
+  <types>
+    <const name="KEY_MAX_LEN" type="long" value="64"/>
+    <const name="VALUE_MAX_LEN" type="long" value="2048"/>
+    <struct name="KeyValuePair">
+      <member name="id" type="int64" key="true" />
+      <member name="key" type="string" stringMaxLength="KEY_MAX_LEN" key="true" />
+      <member name="value" type="string" stringMaxLength="VALUE_MAX_LEN" />
+	  <member name="time" type="int64" />
```

Comment:
I think you could remove this time field from the information you send. The sample info already has a source timestamp you could use to see when the sample was actually written.

Reply:
The use of the time field here is for when the field was updated, not when the individual message was sent.  I would prefer to leave it in although its not strictly needed for the example.

Reply:
I understand now. You could still supply your own timestamp using this write overload (write_w_timestamp in the traditional APIs):

https://community.rti.com/static/documentation/connext-dds/6.0.1/doc/api/connext_dds/api_cpp2/classdds_1_1pub_1_1DataWriter.html#a4adf03799a3c963b512aeb2b75b794a1

My point was to show that DDS already sends a timestamp in the sample info, so adding a timestamp in the data portion can be redundant -- and you need to calculate two timestamps. If you provide your own you only have to send and calculate that one.

Reply:
I see now.  Ok, well if best practice is to rely on the source_timestamp then I will take out the user defined time field altogether to keep the example focused.

Reply:
Sounds good! Basically you can set {{source_timestamp}} with whatever timestamp you want, so you can use that for encoding the time when the field was updated rather than when it was written by using `write_w_timestamp()` or the `write()` overload where you